endpoint: Fix label replacement

jrajahalme opened the pull request:
Fix bugs in label updates:
- Information labels were never deleted
- Information label assignments were logged even when nothing changed
- Identity label updates never updated the Source or Value of an existing label

Add unit test for label updates.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

joestringer commented on pkg/endpoint/endpoint_test.go:
```diff
@@ -171,6 +172,44 @@ func (s *EndpointSuite) TestEndpointStatus(c *C) {
+	c.Assert(rev, Equals, 0)
+	c.Assert(string(e.OpLabels.OrchestrationIdentity.SortedList()), Equals, "cilium:foo=bar;cilium:zip=zop;")
+	// Remove one label, change the source and value of the other.
+	rev = e.replaceIdentityLabels(pkgLabels.Map2Labels(map[string]string{"foo": "zop"}, "ngnix"))
```
`nginx` is misspelled ;-)

borkmann replied:
@jrajahalme Could you still fix this before merge? Thx

jrajahalme replied:
Fixed.

joestringer reviewed (approved):
Looks good to me.

jrajahalme committed:
endpoint: Fix label replacement.

Fix bugs in label updates:
- Information labels were never deleted
- Information label assignments were logged even when nothing changed
- Identity label updates never updated the Source or Value of an existing label

Add unit test for label updates.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
